src/anyio/_backends/_asyncio.py

```diff
@@ -751,12 +732,16 @@ async def __aexit__(
                 if not isinstance(exc_val, CancelledError):
                     self._exceptions.append(exc_val)
 
+            loop = get_running_loop()
             try:
                 if self._tasks:
                     with CancelScope() as wait_scope:
                         while self._tasks:
+                            if self._on_completed_fut is None:
+                                self._on_completed_fut = loop.create_future()
```

Comment:
Is there any case where we _don't_ want to replace the future here?

Reply:
Oh yeah, we can always set it

Reply:
```suggestion
                            self._on_completed_fut = loop.create_future()
```